rust/src/runner/context.rs

```diff
@@ -117,6 +99,28 @@ impl TestContext {
             resume_unwind(e)
         }
     }
+}
+
+impl<const SER: bool> TestContext<SER> {
```

Comment:
Does anything currently enforce that `as_user` cannot be called from a non-serialized test case?  I suggest enforcing it via the type system, like this:
```rust
struct TestContext {}
impl TestContext {
    fn create_file() {...}
    ...
}
struct SerializedTestContext(TestContext)
impl SerializedTestContext {
    fn create_file(...) {self.0.create_file(...)}
    ...
    fn as_user(...) {...}
}
```
This way `TestContext::as_user` won't even exist, so you'll get a compile failure if you try to change user from within a non-serialized test.  You'll also probably need to `impl Deref<Target=TestContext> for SerializedTestContext` so methods like `assert_ctime_changed` can just take a `&TestContext` argument.

Reply:
Yes, that is the purpose of the const generic. It ensures that only tests which accepts `SerializedTestContext` (e.g. `TestContext<true>`) can use serialized functions (which is also why the utimesnat tests fail). Although, the newtype + deref  pattern should definitely work too (and is more idiomatic).